core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -753,21 +751,25 @@ class HttpGoogleServicesDAO(
     } )
   }
 
-  override def getGenomicsOperation(jobId: String): Future[Option[JsObject]] = {
-
-    import spray.json._
-
+  override def getGenomicsOperation(jobId: String): Future[JsObject] = {
     val opId = s"operations/$jobId"
-
     val genomicsApi = new Genomics.Builder(httpTransport, jsonFactory, getGenomicsServiceAccountCredential).setApplicationName(appName).build()
     val operationRequest = genomicsApi.operations().get(opId)
 
-    retryWithRecoverWhen500orGoogleError[Option[JsObject]](() => {
-      // Google library returns a Map[String,AnyRef], but we don't care about understanding the response
-      // So, use Google's functionality to get the json string, then parse it back into a generic json object
-      Some(executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject)
-    }) {
-      case t: HttpResponseException if t.getStatusCode == StatusCodes.NotFound.intValue => None
+    // This predicate is true when it's a 500 or Google error but NOT a Google 404.
+    //
+    // This uses the "applicative builder" syntax from cats to combine 2 predicates
+    // (when500orGoogleError and whenGoogle404) using a binary operator (e.g.
+    // (Boolean, Boolean) => Boolean), and return a resulting predicate.
+    val predicate = (when500orGoogleError |@| whenGoogle404).map((a, b) => a && !b)
```

Comment:
I hope you guys aren't too annoyed at me using cats again. :)

In this case I think it's cleaner: combining predicates using `|@|` and passing the combined predicate to the retry loop; rather than hard-coding the predicate, jamming a `recover` on the Future, and using `Option` to represent a "special" recoverable value not covered by the predicate.